### ISSUE TITLE
The cache file is too large

### ISSUE THREAD

**Issue by shanghaiyangming** [shaypal5/cachier#29]
If a function is repeatedly called with different parameters, I find that the file in the cache directory still has only one file named after the function name. This leads to a huge file size when the function is called many times, so it reads
And the speed of rewriting is not as good as when the cache is not used.

**Comment by shaypal5** [comment]
Hmm. Understandable.

I never really got into dealing with cleaning old cache values, or with dealing with such cases.

Do you want to perhaps contribute code to solve this problem?

**Comment by erap129** [comment]
I think a straightforward solution is to create a different file for each parameter set used to call the function (can make this optional maybe). Would be happy to contribute if you think this is a good idea.

**Comment by shaypal5** [comment]
First, we need to make sure there's no performance issue here. At the moment, if I remember correctly, the single-file-per-function-scheme might make things faster by not opening and acquiring locks on different files. I'm not sure if that's the case, since the pickle cache has some intermediate layer in-memory, but there also a different thread opened for checking and reading from cache.

So, anyway, first thing is reading through the code and understanding how this affects the amount of disk reads.

Secondly, we need to benchmark the change (or lack-thereof, if believe this doesn't impede performance). 

Finally, if this does work properly, we also need  a naming scheme that doesn't break for lots of parameters, or long name parameters. Maybe a constant 

I would LOVE a contribution by you, @erap129 , so please go ahead with examining these issues.

**Comment by eladrap** [comment]
actually I am trying to get started with the project, and I can't seem to get the tests running:

`(cachier_env) C:\Users\eladr\Documents\GitHub\cachier>python -m pytest
ImportError while loading conftest 'C:\Users\eladr\Documents\GitHub\cachier\tests\conftest.py'.
tests\conftest.py:6: in <module>
    from .test_mongo_core import _test_mongetter
tests\test_mongo_core.py:71: in <module>
    ???
cachier\core.py:128: in cachier
    core = _MongoCore(mongetter, stale_after, next_time)
cachier\mongo_core.py:47: in __init__
    self.mongo_collection = self.mongetter()
tests\test_mongo_core.py:56: in _test_mongetter
    _test_mongetter.client = _get_cachier_db_mongo_client()
tests\test_mongo_core.py:40: in _get_cachier_db_mongo_client
    host = quote_plus(CFG[CfgKey.HOST])
..\..\..\Anaconda3\envs\cachier_env\lib\site-packages\birch\core.py:276: in __getitem__
    val = self._getitem_helper(key, self._val_dict)
..\..\..\Anaconda3\envs\cachier_env\lib\site-packages\birch\core.py:268: in _getitem_helper
    raise KeyError("{}: No configuration value for {}.".format(
E   KeyError: 'cachier: No configuration value for TEST_HOST.'`  
  
This happens also when running any non-mongo related tests. Do I need to add certain system variables?

**Comment by shaypal5** [comment]
No, that's ok. Mongo tests are supposed to work locally. They will be run by Travis when you submit a PR.

But it does seem like we'll have to set up some kind of memory-local mock so you can run all other tests.
Set the following environment variables with dummy values and let me know if that helps:
![image](https://user-images.githubusercontent.com/917954/116775624-b2ed0600-aa6c-11eb-9fc9-2f699c908d80.png)

**Comment by shaypal5** [comment]
@erap129  

It seems like you tried to fix `test_delete_cache_file`, however it still fails:
https://github.com/shaypal5/cachier/runs/3612547559?check_suite_focus=true

It also fails on `master`. Let's and figure this out.
